public/playersBetController.js

```diff
@@ -0,0 +1,147 @@
+    );
+  }
+
+  initPlayers() {
+    this.amountOfPlayers = this.players.length;
+    this.updateActivePlayer();
+  }
+
+  selectDomElements() {
+    this.betControls = document.querySelector(".bet-controls");
+    this.bet25 = this.controlsContainerEle.querySelector(".btn--25");
+    this.bet50 = this.controlsContainerEle.querySelector(".btn--50");
+    this.bet100 = this.controlsContainerEle.querySelector(".btn--100");
+    this.clear = this.controlsContainerEle.querySelector(".btn--clear");
+    this.currentPotCont =
+      this.controlsContainerEle.querySelector(".bet-controls__pot");
+    this.curretBetCont =
+      this.controlsContainerEle.querySelector(".bet-controls__bet");
+    this.confirm = this.controlsContainerEle.querySelector(".btn--bet");
+  }
+
+  addEventListeners() {
+    this.bet25.addEventListener("click", this.betting.bind(this));
+    this.bet50.addEventListener("click", this.betting.bind(this));
+    this.bet100.addEventListener("click", this.betting.bind(this));
+    this.clear.addEventListener("click", this.clearBet.bind(this));
+    this.confirm.addEventListener("click", this.confirmBet.bind(this));
+  }
+
+  updateActivePlayer() {
+    this.activePlayer = this.players[this.currentPlayerTurn];
+    document
+      .querySelector(`#${this.activePlayer.name} .card`)
+      .classList.add("active");
+  }
+
+  nextPlayer() {
+    document
+      .querySelector(`#${this.activePlayer.name} .card`)
+      .classList.remove("active");
+    if (this.amountOfPlayers - 2 === this.currentPlayerTurn) {
+      this.betControls.remove();
+      this.playersStillBetting = true;
+      return;
+    } else {
+      this.currentPlayerTurn++;
+      this.updateActivePlayer();
+      this.currentPotCont.innerHTML = `Pot: ${this.activePlayer.pot}`;
+      this.curretBetCont.innerHTML = `Current bet: ${this.currentBet}`;
+    }
+  }
+
+  betting(e) {
+    this.clear.disabled = false;
+    this.confirm.disabled = false;
+    const betAddition = e.target.innerText;
+    if (this.activePlayer.pot - betAddition < 0) {
+      return;
+    }
+    this.activePlayer.pot -= betAddition;
+    this.currentBet += 1 * betAddition;
+    this.currentPotCont.innerHTML = `Pot: ${this.activePlayer.pot}`;
+    this.curretBetCont.innerHTML = `Current bet: ${this.currentBet}`;
+  }
+
+  clearBet() {
+    this.clear.disabled = true;
+    this.confirm.disabled = true;
+    this.activePlayer.pot += this.currentBet;
+    this.currentBet = 0;
+    this.currentPotCont.innerHTML = `Pot: ${this.activePlayer.pot}`;
+    this.curretBetCont.innerHTML = `Current bet: ${this.currentBet}`;
+  }
+
+  confirmBet() {
+    this.clear.disabled = true;
+    this.confirm.disabled = true;
+    let playerHtmlCont = document.querySelector(`#${this.activePlayer.name}`);
+    playerHtmlCont.insertAdjacentHTML(
+      "afterbegin",
+      `Betting: ${this.currentBet}`
+    );
+    this.currentBet = 0;
+    this.nextPlayer();
+  }
+
+  async allowInitialBet(fn) {
```

Comment:
The loop seems unnecessary because you are already tracking if all players have placed their bet in like `87` of this bet controller, therefore you can start the game from there when all players have placed their bet